test/test_signalfd.rs

```diff
@@ -11,8 +11,7 @@ use nix::unistd;
 fn main() {
     print!("test test_signalfd ... ");
 
-    let mut mask = signal::SigSet::empty();
-    mask.add(signal::SIGUSR1).unwrap();
+    let mask: signal::SigSet = signal::SIGUSR1.into();
```

Comment:
I tried to figure out how to do this and I've bumped up against the edges of my knowledge of Rust. Because `SignalFd::new` (and the other various methods for `SignalFd`) take `&SigSet`, we can't easily do something like this:

```
let sigmask = Signal::SIGTERM;
SignalFd::new(sigmask);
// or
SignalFd::new(&sigmask);
```

`impl From<&Signal> for &SigSet` won't work because of trait specialization not being implementated, but there's an even deeper issue. I was trying to get `impl AsRef<SigSet> for Signal` but realised that it doesn't appear to be possible (given Rust's borrowing model) to return an immutable reference to something allocated in the function that you're returning from. As a simple example, this simply doesn't work for obvious reasons:

```rust
impl AsRef<SigSet> for Signal {
    fn as_ref(&self) -> &SigSet {
        &self.clone().into()
    }
}
```

Is there a way to make that work? It seems quite silly that you can't convert `&Into<T>` into `Into<&T>` but there's probably some much harder problem here than I'm seeing.